docs/future/compatibility.md

```diff
@@ -0,0 +1,174 @@
+  // exactly as it was passed to the component.
+  props: object
+  
+  // The backing instance to the node. Can be null in the case of "host" nodes and SFCs.
+  // React alternative libs can choose to provide whatever concept here applies.  If it is
+  // provided, Enzyme will expect it to follow the ComponentInterface type specified below
+  instance: ComponentInstance?
+  
+  // This is a transformed version of `props.children`. It is an array or null. If it's an array,
+  // it's an array of Nodes, meaning that they are LiteralNodes or RSTNodes. This means that 
+  // non-rendering values such as `false`, `null`, and `[]` wouldn't make it in here. This is also
+  // a "flat" array.
+  children: [Node]?
+  
+  // This can feel similar to children, but it's not the same thing at all. For a given node,
+  // this corresponds to the result of the `render` function with the provided props, but transformed
+  // into an RST.  For "host" nodes, this will always be `null`.
+  rendered: Node?
+|}
+```
+
+
+A `ComponentInstance` is going to be expected by enzyme to follow a certain interface in order
+for certain API methods to actually work, but they don't *need* to. Those methods just won't be
+guaranteed to work in those cases.
+
+
+So far, that interface is:
+```js
+type ComponentInstance = {
+  state: object?
+  context: object?
+  setState: function
+}
+```
+
+
+### Enzyme Adapter Protocol
+
+**Definitions:**
+
+An `Element` is considered to be whatever data structure is returned by the JSX pragma being used. In the 
+react case, this would be the data structure returned from `React.createElement`
+
+
+```js
+type RendererOptions = {
+  // I'm not 100% sure about this, but I think with an API like this, we could implement
+  // `mount` and `shallow` with the same renderer.
+  ?isLeaf(Element): bool
+}
+
+type EnzymeAdapter = {
+  // Provided a bag of options, return an `EnzymeRenderer`. Some options can be implementation
+  // specific, like `attach` etc. for React, but not part of this interface explicitly.
+  createRenderer(RendererOptions?): EnzymeRenderer
+
+  // converts an RSTNode to the corresponding JSX Pragma Element. This will be needed
+  // in order to implement the `Wrapper.mount()` and `Wrapper.shallow()` methods, but should
+  // be pretty straightforward for people to implement.
+  nodeToElement(RSTNode): Element
+}
+
+type EnzymeRenderer = {
+  // both initial render and updates for the renderer.
+  render(Element): void
+  
+  // retrieve a frozen-in-time copy of the RST.
+  getNode(): RSTNode?
+}
+```
+
+
+### Using different adapters with Enzyme
+
+At the top level, Enzyme would expose a `configure` method, which would allow for an `adapter`
+option to be specified and globally configure Enzyme's adapter preference:
+
+```js
+import Enzyme from 'enzyme';
+import ThirdPartyEnzymeAdapter from 'third-party-enzyme-adapter';
+
+Enzyme.configure({ adapter: ThirdPartyEnzymeAdapter });
```

Comment:
I'm more partial to the explicit API as well. It also makes it really clear how to use different adapters in the same test suite.  Though I am kind of bummed we will be joining the ranks of tools that don't "just work" with one npm install :/